go/base/context.go

```diff
@@ -214,6 +215,8 @@ type ContextConfig struct {
 	}
 }
 
+var Context *MigrationContext
```

Comment:
I'd like to get rid of this global variable.

Reply:
The global variable is used in base/utils.go ValidateConnection. I need to get the aliyunRDS flag to avoid get the port from database. If don't use the global variable, add a param ( migrationContext may be ) to the function? or any other suggestions ?

Reply:
Indeed it is used there. Sorry, I've missed this.

I still wish to get rid of the global variable. It will not play well with testing and embedding of `gh-ost` as a library. Please add a `migrationContext *MigrationContext` argument to base/utils.go `ValidateConnection()` function and use it.

Reply:
Branch is updated, thanks for the review.